tests/lib/bin/version-tool

```diff
@@ -0,0 +1,323 @@
+
+    def test_lt(self):
+        # type: () -> None
+        self.assertTrue(consistent_relation("<", -1))
+        self.assertFalse(consistent_relation("<", 0))
+        self.assertFalse(consistent_relation("<", +1))
+
+    def test_le(self):
+        # type: () -> None
+        self.assertTrue(consistent_relation("<=", -1))
+        self.assertTrue(consistent_relation("<=", 0))
+        self.assertFalse(consistent_relation("<=", +1))
+
+    def test_unknown(self):
+        # type: () -> None
+        with self.assertRaises(ValueError):
+            consistent_relation("???", 0)
+
+
+def numeric_part_of(text):
+    # type: (Text) -> int
+    """
+    numeric_part_of returns the integer representing the numeric portion of text.
+
+    If text does not start with an decimal integer the result is zero. The
+    function never fails.
+    """
+    match = re.match("^([0-9]+)", text)
+    if match:
+        return int(match.group(1))
+    return 0
+
+
+class NumericPartOfTests(unittest.TestCase):
+    def test_simple(self):
+        # type: () -> None
+        self.assertEqual(numeric_part_of("12"), 12)
+        self.assertEqual(numeric_part_of("12-stuff"), 12)
+        self.assertEqual(numeric_part_of("stuff-12"), 0)
+        self.assertEqual(numeric_part_of("-12"), 0)
+
+
+def naive_version_cmp(a, b):
+    # type: (Text, Text) -> int
+    """
+    naive_version_cmp naively compares two version numbers.
+
+    The algorithm considers each version to be a tuple of integers. Non,
+    integer elements or element fragments are discarded.
+
+    Comparison is performed on by considering the leftmost element in each
+    tuple. First pair of numbers that are not equal determine the result of the
+    comparison. Tuples have unequal length then missing elements are
+    substituted with zero.
+
+    The return value is 0 if the version strings are equal, -1 if version a is
+    smaller or +1 if version b is smaller.
+    """
+    a_items = a.split(".")
+    b_items = b.split(".")
+    if PY2:
+        zip_longest_fn = itertools.izip_longest
+    else:
+        zip_longest_fn = itertools.zip_longest
+    for a_item, b_item in zip_longest_fn(a_items, b_items, fillvalue="0"):
+        a_val = numeric_part_of(a_item)
+        b_val = numeric_part_of(b_item)
+        delta = a_val - b_val
+        if delta != 0:
+            return 1 if delta > 0 else -1
+    return 0
+
+
+class NaiveVersionCmpTests(unittest.TestCase):
+    def test_simple(self):
+        # type: () -> None
+        self.assertEqual(naive_version_cmp("10", "10"), 0)
+        self.assertEqual(naive_version_cmp("10", "20"), -1)
+        self.assertEqual(naive_version_cmp("20", "10"), +1)
+
+    def test_many_segments(self):
+        # type: () -> None
+        self.assertEqual(naive_version_cmp("1.2.3", "1.2.3"), 0)
+        self.assertEqual(naive_version_cmp("1.2.3", "1.3.4"), -1)
+        self.assertEqual(naive_version_cmp("1.4.3", "1.2.3"), +1)
```

Comment:
I think you should also add test cases where the first elements of versions is the same and the penultimate elements of versions are different, respectively i.e.
```python
        self.assertEqual(naive_version_cmp("1.0.0","1.1.0"), -1)
        self.assertEqual(naive_version_cmp("0.1.2","1.1.2"), -1)
```
I've seen naive version comparison algorithms get confused by these kinds of cases

Reply:
Added